harness/determined/_trial_controller.py

```diff
@@ -44,11 +70,14 @@ def __init__(
         # The only time that workloads should be non-None here is unit tests or test mode.
         self.workloads = workloads
 
-        self.prof = profiler.ProfilerAgent.from_env(
-            env,
-            global_rank=context.distributed.rank,
-            local_rank=context.distributed.local_rank,
-        )
+        if hasattr(context._core.train, "_session"):
+            # XXX: stealing this session feels _horrible_
+            sess = context._core.train._session
+            self.prof = _profiler_agent_from_env(
+                sess, env, context.distributed.rank, context.distributed.local_rank
+            )
+        else:
+            self.prof = profiler.DummyProfilerAgent()
```

Comment:
Highlighting this as one of the questionable parts of the PR.

Reply:
Mmm. Doesn't feel great to me, either.

Trying to unwind...

A `ProfilerAgent` used to, say, `get_trial_profiler_available_series`, which itself
1. constructed a URL from a passed master address
2. used `api.get` to GET that URL.

Now, we no longer use `api.get`, we use `*Session.get` and we need a `Session`. Hence the grossness.

But I don't think we need a `Session`, per se, because the old `get` wasn't signed. We'll be good with an `UnauthorizedSession`, instead? And that, we can create from the `env.master_url` here right in this `__init__`, no `context._core.train._session` needed.

Are all those claims correct?

Reply:
no, they are not correct.

They were using the cli_auth and cli_certs to "magically" authenticate in the old `do_request()`, which is why there was that `# TODO: refactor profiler to not use cli_certs` line somewhere that you commented on.